agent/dns.go

```diff
@@ -195,7 +202,7 @@ func (d *DNSServer) handleQuery(resp dns.ResponseWriter, req *dns.Msg) {
 	// Setup the message response
 	m := new(dns.Msg)
 	m.SetReply(req)
-	m.Compress = !d.config.DisableCompression
+	m.Compress = !d.disableCompression.Load().(bool)
```

Comment:
In places like this where you call load on an atomic value, are you guaranteed that Store() has been called previously? Otherwise this statement panics. To be defensive I would check if `d.disableCompression.Load() != nil` first before doing the .(bool)